packages/calcite-components/src/components/list/list.tsx

```diff
@@ -960,6 +981,16 @@ export class List
     }
   };
 
+  private handleInteractionModeWarning(): void {
+    if (
+      this.interactionMode === "static" &&
+      this.selectionMode !== "none" &&
+      this.selectionAppearance === "border"
+    ) {
+      console.warn(`selection-appearance=“border” requires interaction-mode=“interactive”`);
```

Comment:
Aaron, Franco and me went over the issue, and landed on setting a detailed JSDoc for documentation while also logging the warning message. But maybe this can be reconsidered? 
cc @jcfranco @ashetland

Reply:
I defer to @jcfranco and @driskull.

Reply:
Great question. Since this is something teams have been asking for, let’s keep the warning in place to ensure developers use the proper configuration. After #10310, we can explore showing this and related warnings (e.g., [color-picker’s incompatible value + format mode](https://github.com/Esri/calcite-design-system/blob/dev/packages/calcite-components/src/components/color-picker/color-picker.tsx#L1133)) only in dev builds, as suggested in [#10038](https://github.com/Esri/calcite-design-system/issues/10038).